app/src/main/java/com/li/weatherapp/data/source/remote/HistoryRemoteDataSource.kt

```diff
@@ -0,0 +1,45 @@
+package com.li.weatherapp.data.source.remote
+
+import android.text.format.DateUtils
+import com.li.weatherapp.data.model.History
+import com.li.weatherapp.data.source.HistoryDataSource
+import com.li.weatherapp.data.source.remote.api.APIQueries
+import com.li.weatherapp.data.source.remote.utils.RemoteAsyncTask
+import com.li.weatherapp.data.source.remote.utils.makeNetworkCall
+import com.li.weatherapp.data.source.utils.OnDataLoadCallback
+import com.li.weatherapp.utils.Constants
+import org.json.JSONObject
+
+class HistoryRemoteDataSource private constructor() : HistoryDataSource {
+
+    override fun getHistory(lat: String, lon: String, callback: OnDataLoadCallback<List<History>>) {
+        RemoteAsyncTask(callback) {
+            getHistoryList(lat, lon)
+        }.execute()
+    }
+
+    private fun getHistory(lat: String, lon: String, dayQuantity: Int): History {
+        val currentTime = System.currentTimeMillis() / DateUtils.SECOND_IN_MILLIS
+        val jsonString = makeNetworkCall(
+            APIQueries.queryHistory(
+                lat,
+                lon,
+                (currentTime - Constants.SECOND_IN_DAY * dayQuantity).toString()
+            )
+        )
+        return JSONObject(jsonString).getJSONObject(History.CURRENT).let(::History)
```

Comment:
sao không viết theo chiều xuôi mà lại viết theo chiều ngược thế ? 
```suggestion
        return History(JSONObject(jsonString).getJSONObject(History.CURRENT))
```